presto-main/src/main/java/com/facebook/presto/spiller/PartitioningSpiller.java

```diff
@@ -28,13 +28,21 @@
 {
     /**
      * Partition page and enqueue partitioned pages to spill writers.
-     * PartitioningSpillResult::isBlocked returns completed future when finished.
+     * Result's @{link PartitioningSpillResult#getSpillingFuture} is completed when spilling is finished.
+     * <p>
+     * This method may not be called if previously initiated spilling is not finished yet.
      */
     PartitioningSpillResult partitionAndSpill(Page page, IntPredicate spillPartitionMask);
 
     /**
-     * Returns iterator of previously spilled Pages from given partition.
+     * Returns iterator of previously spilled pages from given partition. Whether partition
+     * can be read more than once, is an implementation detail.
```

Comment:
Because I don't want to guarantee exception in the repeated-read case so that there is possibility to change this behavior for the sake of some future functionality that simply requires repeated reads.

Reply:
Given this description, a caller must be defensive and treat this method as single use.  Right?  If so, his comes down to if we guarantee that they get an exception or not when they violate that contract.  If that is the case, simply say something like `Callers are expected to call this method once.  Calling multiple times can results in undefined behavior`